ext/openssl/ossl.c

```diff
@@ -1150,9 +1150,29 @@ Init_openssl(void)
     /*
      * Version number of OpenSSL the ruby OpenSSL extension was built with
      * (base 16)
+     *
+     * OpenSSL 3: 0xMNN00PP0 (major minor 00 patch 0)
+     * OpenSSL 1: 0xMNNFFPPS (major minor fix patch status)
+     * LibreSSL : 0x20000000 (fixed value)
```

Comment:
RDoc won't parse this well. This should be converted to a simple list or a labeled list (https://ruby.github.io/rdoc/RDoc/MarkupReference.html#class-RDoc::MarkupReference-label-Labeled+Lists).

Personally, I prefer calling the second format as "OpenSSL before 3" or "OpenSSL 1.1(.1) or before". While it is not wrong, the major number was practically considered to be the "1.1" part.

This is somewhat similar to Ruby: I don't think "Ruby 2" is a popular term to use to refer to Ruby 2.0.0-2.7.

Reply:
I rebased this PR. I used a labeled list. I also used the monofont <https://ruby.github.io/rdoc/RDoc/MarkupReference.html#class-RDoc::MarkupReference-label-Monofont> to describe the format. Because OpenSSL 1.0 or 1.1's manual for OPENSSL_VERSION_NUMBER <https://www.openssl.org/docs/man1.0.2/man3/OPENSSL_VERSION_NUMBER.html> is using a code block to describe the format. I thought it's natural to use it to describe the format.